src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -200,39 +199,25 @@ private void configureHostChecking(Properties hash) {
 
   private void configureProxy(ClientSession session) {
     if (proxyConfig != null) {
-
-      // Proxy proxy = null;
+      InetSocketAddress proxyAddress = new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort());
+      InetSocketAddress remoteAddress = new InetSocketAddress(this.host, this.port);
       switch (proxyConfig.getProtocol()) {
         case HTTP:
-          // ProxyHTTP proxyHttp = new ProxyHTTP(proxyConfig.getHost(), proxyConfig.getPort());
-          // if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
-          // proxyHttp.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
-          // }
-          // proxy = proxyHttp;
-          break;
-
-        case SOCKS4:
-          // ProxySOCKS4 proxySocks4 = new ProxySOCKS4(proxyConfig.getHost(), proxyConfig.getPort());
-          // if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
-          // proxySocks4.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
-          // }
-          // proxy = proxySocks4;
+          session.setClientProxyConnector(proxyConfig.getUsername() != null && proxyConfig.getPassword() != null
+              ? new HttpClientConnector(proxyAddress, remoteAddress, proxyConfig.getUsername(),
+                                        proxyConfig.getPassword().toCharArray())
+              : new HttpClientConnector(proxyAddress, remoteAddress));
           break;
-
         case SOCKS5:
-          // ProxySOCKS5 proxySocks5 = new ProxySOCKS5(proxyConfig.getHost(), proxyConfig.getPort());
-          // if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
-          // proxySocks5.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
-          // }
-          // proxy = proxySocks5;
-          // break;
-
+          session.setClientProxyConnector(proxyConfig.getUsername() != null && proxyConfig.getPassword() != null
+              ? new Socks5ClientConnector(proxyAddress, remoteAddress, proxyConfig.getUsername(),
+                                          proxyConfig.getPassword().toCharArray())
+              : new Socks5ClientConnector(proxyAddress, remoteAddress));
```

Comment:
This code is pretty confusing. Perhaps the inline condition is not a good fit and makes it more cryptic. If the proxyConfig contains both user and password it calls to a different constructor method. However it makes absolutely no difference, since both the constructor without user and password ends up calling the other one with the user and password `null`. Just check for null on the password so as to turn it into the char array and call the constructor with all the parameters.